packages/oscal-react-library/src/components/HoverablePopover.tsx

```diff
@@ -0,0 +1,90 @@
+import { Box, Popover } from "@mui/material";
+import React from "react";
+import { OSCALMarkupLine, OSCALMarkupMultiLine } from "./OSCALMarkupProse";
+
+export interface HoverablePopoverProps {
+  /**
+   * The content of the popover.
+   */
+  popoverContent: React.ReactNode;
+
+  /**
+   * Content to hover over.
+   */
+  children: React.ReactNode;
+}
+
+export const HoverablePopover: React.FC<HoverablePopoverProps> = ({ popoverContent, children }) => {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
+
+  const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handlePopoverClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+
+  return (
+    <>
+      <Box
+        component="span"
+        aria-owns={open ? "mouse-over-popover" : undefined}
+        aria-haspopup="true"
+        onMouseEnter={handlePopoverOpen}
+        onMouseLeave={handlePopoverClose}
+      >
+        {children}
+      </Box>
+      <Popover
+        open={open}
+        sx={{
+          pointerEvents: "none",
+          maxWidth: "30em",
+        }}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        anchorEl={anchorEl}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+        onClose={handlePopoverClose}
+        disableRestoreFocus
+      >
+        <Box sx={{ padding: "1em" }}>{popoverContent}</Box>
+      </Popover>
+    </>
+  );
+};
+
+export interface MarkupPopoverProps {
+  /**
+   * The markup multi line content of the popover.
+   */
+  popoverContent: string;
+
+  /**
+   * Content to hover over.
+   */
+  children: React.ReactNode;
+}
+
+export const MarkupMultiLinePopover: React.FC<MarkupPopoverProps> = ({
+  popoverContent,
+  children,
+}) => {
+  const markup = <OSCALMarkupMultiLine>{popoverContent}</OSCALMarkupMultiLine>;
+
+  return <HoverablePopover popoverContent={markup}>{children}</HoverablePopover>;
+};
+
+export const MarkupLinePopover: React.FC<MarkupPopoverProps> = ({ popoverContent, children }) => {
+  const markup = <OSCALMarkupLine>{popoverContent}</OSCALMarkupLine>;
+
+  return <HoverablePopover popoverContent={markup}>{children}</HoverablePopover>;
+};
```

Comment:
I don't understand how/why this gives much benefit over the callers doing this themselves. It feels like an awkward API. The benefit is a `string` instead of a `ReactNode`? But if they need to pass additional props, they have to fallback to `HoverablePopover` anyway?

Reply:
I don't _love_ having to pass a component as a prop and this seemed like such a common use case (maybe the only use case) that the wrapper felt beneficial. 

I had considered making `HoverablePopover` private and just using those wrappers since that was the main case the component would be used.

I can remove it if we feel strongly they don't provide much value.

Reply:
I would almost suggest just checking the `popover` type to see if it's a `string` and format it differently with `OSCALMarkupLine`. This would avoid the need to create a second `interface` and `const`.

Reply:
> I would almost suggest just checking the `popover` type to see if it's a `string` and format it differently with `OSCALMarkupLine`. This would avoid the need to create a second `interface` and `const`.

If a string is being dispalyed, it is not always going to be `markup`. Even if it was, it may be a `markup line` or a `markup multiline`.

Reply:
Sorry, I understand now. Maybe I'm just not understanding all the cases of how a multiline will be provided, but couldn't we just check the `string` to have a newline character in it to determine if it should be `markup multiline`?

Reply:
`"a *cool* string"` is valid both as markup line and markup multi line. There are places in the spec that very specifically state which we should choose. So I don't think we can do that as nice as it would be. 

I think the issue Kyle is mentioning is the inclusion of those wrapper components at all. As you _can_ just put the correct mark up component in the prop.